readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl

```diff
@@ -134,6 +134,6 @@ else:
 
 # Add custom RTD extension
 if 'extensions' in globals():
-    extensions.append("readthedocs_ext.readthedocs")
+    extensions.insert(0, "readthedocs_ext.readthedocs")
```

Comment:
I don't understand either why in this way it works (or not), but we definetely need to add a comment here explaining why we are changing this since we don't have an easy way to write a test.

Reply:
Also, there isn't a good way of explaining it :grin:, maybe a link to this PR is enough?